Src/HALAL/Services/Communication/I2C/I2C.cpp

```diff
@@ -0,0 +1,248 @@
+/*
+ * SPI.hpp
+ *
+ *  Created on: 5 nov. 2022
+ *      Author: Pablo
+ */
+#include "Communication/I2C/I2C.hpp"
+
+#ifdef HAL_I2C_MODULE_ENABLED
+
+unordered_map<uint8_t, I2C::Instance* > I2C::active_i2c;
+
+uint16_t I2C::id_counter = 0;
+
+optional<uint8_t> I2C::inscribe(I2C::Peripheral& i2c){
+	if ( !I2C::available_i2cs.contains(i2c)){
+
+		return nullopt; //TODO: Error handler or throw the exception
```

Comment:
```suggestion
		return nullopt;
```
en los inscribes quedamos que el error se propaga